index.js

```diff
@@ -41,4 +41,92 @@ form.addEventListener('submit', (event) => {
   const email = form.elements.mail;
   const emailText = email.value;
   emailValidation(emailText, invalidEmail, event);
-});
\ No newline at end of file
+});
+// preserve-data-in-the-browser
+
+// get form elements and distract
+const Form = document.getElementById('form');
+const {
+  name: nameInput,
+  email: emailInput,
+  message: messageInput,
+} = Form.elements;
+
+// check local storage available
+// - if available : create local storage object
+// - if not: null
+
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return (
+      e instanceof DOMException
+      // everything except Firefox
+      && (e.code === 22
+        // Firefox
+        || e.code === 1014
+        // test name field too, because code might not be present
+        // everything except Firefox
+        || e.name === 'QuotaExceededError'
+        // Firefox
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+      // acknowledge QuotaExceededError only if there's something already stored
+      && storage
+      && storage.length !== 0
+    );
+  }
+}
+
+let availableStorage;
+
+if (storageAvailable('localStorage')) {
+  // Yippee! We can use localStorage awesomeness
+  availableStorage = window.localStorage;
+} else {
+  // Too bad, no localStorage for us
+  availableStorage = null;
+}
+// create a single object for the data
+const formDataObject = {};
+
+function storeData() {
+  formDataObject.name = nameInput.value;
+  formDataObject.email = emailInput.value;
+  formDataObject.message = messageInput.value;
+  const jsonData = JSON.stringify(formDataObject);
+  availableStorage.setItem('FormData', jsonData);
+}
+// listen to change on input fields
+nameInput.addEventListener('change', () => {
+  storeData();
+});
+
+emailInput.addEventListener('change', () => {
+  storeData();
+});
+
+messageInput.addEventListener('change', () => {
+  storeData();
+});
+// retrive the data by stringify and
+// set the object to local on loads of page
+function retrieveData() {
+  const contactData = availableStorage.getItem('FormData');
+  const parseContactData = JSON.parse(contactData);
+  if (contactData?.length > 0) {
+    const { name, email, message } = parseContactData;
+    nameInput.value = name || '';
+    emailInput.value = email || '';
+    messageInput.value = message || '';
+  }
+}
```

Comment:
Dear @fithamlak, Good job so far, As part of the main requirement for this project we should be able to store and retrieve data from local storage. Your form is saving the data only once and only if hit submit. on the next reload the content are not available. Please address this for a better user experience. 
### Remember the following point as you work on the contact form. 

- [ ] The contact form information is persistent to page reloads, This includes after filling the form and not submitting it.
- [ ] When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page. and If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.
- [ ] To achieve both features that I have mentioned you could make use of `localStorage.getItem` and `localStorage.setItem`